services/management/dbaas/components_service_test.go

```diff
@@ -507,34 +507,177 @@ func TestFilteringOutOfUnsupportedVersions(t *testing.T) {
 	})
 }
 
-func TestCheckForOperatorUpdate(t *testing.T) {
-	setup := func(t *testing.T, clusterName string, response *VersionServiceResponse, port string) (dbaasv1beta1.ComponentsServer, *mockDbaasClient) {
-		t.Helper()
+const (
+	onePointEight       = "1.8.0"
+	defaultPXCVersion   = "5.7.26-31.37"
+	latestPXCVersion    = "8.0.0"
+	defaultPSMDBVersion = "3.6.18-5.0"
+	latestPSMDBVersion  = "4.5.0"
+	port                = "5497"
+	clusterName         = "installoperator"
+)
 
-		uuid.SetRand(new(tests.IDReader))
+func setup(t *testing.T, clusterName string, response *VersionServiceResponse, port, defaultPXC, defaultPSMDB string) (*reform.Querier, dbaasv1beta1.ComponentsServer, *mockDbaasClient) {
+	t.Helper()
 
-		sqlDB := testdb.Open(t, models.SetupFixtures, nil)
-		db := reform.NewDB(sqlDB, postgresql.Dialect, reform.NewPrintfLogger(t.Logf))
-		dbaasClient := new(mockDbaasClient)
+	uuid.SetRand(new(tests.IDReader))
 
-		kubernetesCluster, err := models.CreateKubernetesCluster(db.Querier, &models.CreateKubernetesClusterParams{
+	sqlDB := testdb.Open(t, models.SetupFixtures, nil)
+	db := reform.NewDB(sqlDB, postgresql.Dialect, reform.NewPrintfLogger(t.Logf))
+	dbaasClient := new(mockDbaasClient)
+
+	kubernetesCluster, err := models.CreateKubernetesCluster(db.Querier, &models.CreateKubernetesClusterParams{
+		KubernetesClusterName: clusterName,
+		KubeConfig:            "{}",
+	})
+	require.NoError(t, err)
+	kubernetesCluster.Mongod = &models.Component{
+		DefaultVersion: defaultPSMDB,
+	}
+	kubernetesCluster.PXC = &models.Component{
+		DefaultVersion: defaultPXC,
+	}
+	require.NoError(t, db.Save(kubernetesCluster))
+
+	vsc, cleanup := newFakeVersionService(response, port, pxcOperator, psmdbOperator, "pmm-server")
+
+	t.Cleanup(func() {
+		cleanup(t)
+		uuid.SetRand(nil)
+		dbaasClient.AssertExpectations(t)
+		assert.NoError(t, db.Delete(kubernetesCluster))
+		require.NoError(t, sqlDB.Close())
+	})
+	return db.Querier, NewComponentsService(db, dbaasClient, vsc), dbaasClient
+}
+
+func TestInstallOperator(t *testing.T) {
+	pmmversion.PMMVersion = "2.19.0"
+
+	response := &VersionServiceResponse{
+		Versions: []struct {
+			Product        string `json:"product"`
+			ProductVersion string `json:"operator"`
+			Matrix         matrix `json:"matrix"`
+		}{
+			{
+				Product:        pxcOperator,
+				ProductVersion: "1.7.0",
```

Comment:
why `1.7.0` is just a string, but `1.8.0` is a constant?

Reply:
It's because linter complained, but I can make constant out of it, too.

Reply:
Done, now we use constants for 1.6.0, 1.7.0, 1.8.0, 1.9.0.